examples/seq2seq/run_seq2seq.py

```diff
@@ -406,6 +413,12 @@ def preprocess_function(examples):
             ]
 
         model_inputs["labels"] = labels["input_ids"]
+
+        # prepare decoder_input_ids
+        if hasattr(model, "prepare_decoder_input_ids_from_labels"):
+            decoder_input_ids = model.prepare_decoder_input_ids_from_labels(labels=model_inputs["labels"])
```

Comment:
@sgugger 

Just realized this, here `labels` is a list but we need a `tensor` for `prepare_decoder_input_ids_from_labels` method. And can't turn that into a tensor here since it's not padded.

Thinking more about this, I would say we remove the `ignore_pad_token_for_loss` argument, this is rather a confusing name. In the previous script, it was used to specify that we should ignore pad tokens as it is (by setting the `ignore_index` in loss to `pad_token_id` ) rather than replacing them with -100 (because of the label smoothing issues). So in any case the pad tokens were ignored, and I don't think there is any reason to not ignore them.

So IMO we should 
- remove the `ignore_pad_token_for_loss` argument
- set `padding` to `True` when `pad_to_max_length` is False
- replace pad with -100 in the pre-processing function
- and remove the `DataCollatorForSeq2Seq`

Reply:
> set padding to True when pad_to_max_length is False

This will pad to the maximum length of the batches sent to the map method of the dataset, not the maximum length of a batch, so this does not work. `DataCollatorForSeqSeq` is needed until we have the datasets v2 release to do that padding as a transform on the dataset (soon!) but we can't deprecate it just now.

Reply:
Aah, yes. Then do you think we could pass the `prepare_decoder_input_ids_from_labels` method to  `DataCollatorForSeq2Seq`  and prepare the `decoder_input_ids` there ?

Reply:
I'm happy with that solution, yes. We can pass the `model` to `DataCollatorForSeq2Seq` and do the check for the method there (better to pass the object than a function).

Reply:
Hey @sgugger , @patil-suraj 
I am trying to run "run_seq2seq.py" to train mT5 for translation task. I am getting the following error though:

`ImportError: cannot import name 'DataCollatorForSeq2Seq' from 'transformers' (unknown location)`

It seems that DataCollatorForSeq2Seq is already removed from the transformers package, right?

P.S. my transformers version: 4.2.2 installed using pip

Reply:
Hey @Arman-IMRSV,

please use issues to report bugs